### PR TITLE
Attach the output of subtests to the parent test

### PR DESCRIPTION
**Pull request by dmage**
Adds support of t.Run output from Go 1.7.

see also #13537

### CODE REVIEW

File: tools/junitreport/pkg/parser/gotest/data_parser.go [hunk 1]
```diff
@@ -10,13 +10,13 @@ func newTestDataParser() testDataParser {
 	return testDataParser{
 		// testStartPattern matches the line in verbose `go test` output that marks the declaration of a test.
 		// The first submatch of this regex is the name of the test
-		testStartPattern: regexp.MustCompile(`=== RUN\s+(.+)$`),
+		testStartPattern: regexp.MustCompile(`=== RUN\s+([^/]+)$`),
```
Comment (stevekuznetsov) [inline tools/junitreport/pkg/parser/gotest/data_parser.go]: Add a comment as to why this is necessary.

File: tools/junitreport/pkg/parser/gotest/data_parser_test.go [hunk 2]
```diff
@@ -21,12 +21,8 @@ func TestMarksTestBeginning(t *testing.T) {
 			testLine: "=== RUN 1234",
 		},
 		{
-			name:     "url",
```
Comment (stevekuznetsov) [inline tools/junitreport/pkg/parser/gotest/data_parser_test.go]: Why is this not valid anymore?

Comment (dmage) [inline tools/junitreport/pkg/parser/gotest/data_parser_test.go]: I don't know how to produce such output. AFAIK, there is always a function name. Am I wrong?

File: tools/junitreport/pkg/parser/gotest/data_parser.go (hunk 1 above)
Comment (dmage) [inline tools/junitreport/pkg/parser/gotest/data_parser.go]: If there is always a function name, then I don't want to treat start of a subtest as a start of test.

Comment (stevekuznetsov) [inline tools/junitreport/pkg/parser/gotest/data_parser.go]: I understand that, but write it in the code.

File: tools/junitreport/pkg/parser/gotest/data_parser_test.go (hunk 2 above)
Comment (stevekuznetsov) [inline tools/junitreport/pkg/parser/gotest/data_parser_test.go]: Hmm, I can't remember how this was acheived.

**Comment by dmage** [comment]
@stevekuznetsov there is a possibility to have a failed test with all subtests succeed. How should it be represented if we consider it as a TestSuite? Should we always have a TestCase for the parent test? Or should we add it only if it failed, but all subtests succeed?

**Comment by stevekuznetsov** [comment]
Good point, maybe the suite would hold a top-level test case for the overall result? But we're going to get weird timing if we end up double-counting everything.

**Pull request closed without merging**